util/calcrewards.py

```diff
@@ -151,10 +151,15 @@ def _calcRewardsUsd(S_USD, P_USD, rewards_avail_USD: float) -> numpy.ndarray:
                 RF[c, i, j] = S_USD[c, i, j] * P_USD[c, j]  # main formula!
 
     # normalize values
-    RF_norm = RF / numpy.sum(RF)
+    RF_norm = RF / (numpy.sum(RF) + 0.000001)  # avoid divide by zero
 
     # filter negligible values (<0.00001% of total RF), then re-normalize
     RF_norm[RF_norm < 0.000001] = 0.0
+
+    if numpy.sum(RF_norm) == 0.0:
```

Comment:
Put this "if zero" check _before_ the "normalize the values" call. Then we don't need the hacky code of `"+ 0.000001"`.

Reply:
Yes, better to have the check twice than to have the hacky code. You can remove the comment line in both cases, which makes the code more compact.